### ISSUE TITLE
How to Plot and Visualize Feature Maps for Each Layer in a YOLOv11 Mode

### ISSUE THREAD

**Issue by WengYan0619** [ultralytics/ultralytics#18805]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, I'm training a model using YOLOv11 and I've enabled visualization by setting **visualize=True** in my training script. However, I'm not seeing the feature maps from the convolutional layers, which I expected to be part of the visual output. Could you guide me on how to access or enable the visualization of feature maps for each convolutional layer during training? Here's the relevant part of my code:

`from ultralytics import YOLO

if __name__ == "__main__":
    
    model = YOLO("yolo11n.pt", task="detect")  # Load model and make predictions
    # Train the model
    results = model.train(
        data = "C:/Users/User/Desktop/University/Annotated_dataset/data.yaml",  # path to your data configuration file
        epochs=350,                      # number of epochs
        imgsz=640,                      # image size
        batch=32,                       # batch size
        device='0',                      # cuda device (use '0' for GPU, 'cpu' for CPU)
        visualize=True,                 # plot results
    )`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @WengYan0619, thank you for your interest in Ultralytics 🚀! We recommend visiting the [Docs](https://docs.ultralytics.com/) for an overview of features and usage examples, including helpful [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) guides.

If this is a ❓ Question about YOLOv11 feature map visualization, could you please provide more specific details or clarifications regarding your output? The code snippet you shared is very helpful! If you're expecting additional visualizations for layer outputs, please confirm that you're working with the latest version of the `ultralytics` package. To ensure compatibility and access the latest features, upgrade with:

```bash
pip install -U ultralytics
```

For 🐛 Bug Reports, if you believe there's an issue with the `visualize` setting, kindly provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) demonstrating exactly how the behavior deviates from expected functionality. This helps us investigate the issue more effectively.

In the meantime, feel free to explore alternate environments for training and debugging:
- **Free GPU Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

Additionally, you can join our vibrant community for real-time support and discussions 😃:
- [Discord](https://discord.com/invite/ultralytics) 🎧
- [Discourse](https://community.ultralytics.com/)
- [Reddit](https://reddit.com/r/Ultralytics)

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. 

An Ultralytics engineer will review your question shortly and assist you further. Let us know if you have any additional info in the meantime, and thanks again for using Ultralytics! ✨

**Comment by Y-T-G** [comment]
It's not a training argument. It's a prediction argument.
